aten/src/ATen/native/mps/operations/View.mm

```diff
@@ -409,6 +441,177 @@
   return broadcastTensor;
 }
 
+MPSGraphTensor* asStridedLayer_reshapeTransposePattern(MPSGraph *graph, MPSGraphTensor *inputTensor, int dstRank, const IntArrayRef& dstSizes, const IntArrayRef& dstStrides, int offset) {
+  NSUInteger inputVolume = 1;
+  for(NSNumber *val : [inputTensor shape])
+      inputVolume *= [val integerValue];
+
+  // 1. Sort the strides in descending order, and determine the dimension order
+  // to permute the sorted strides back to the requested strides
+  NSMutableArray *dimOrder = [[NSMutableArray new] autorelease];
+  NSMutableArray *sortedTargetStrides = [[NSMutableArray new] autorelease];
+  {
+    for (NSUInteger i = 0; i < dstRank; i++) {
+      dimOrder[i] = [NSNumber numberWithInteger: i];
+      sortedTargetStrides[i] = [NSNumber numberWithInteger: dstStrides[i]];
+    }
+
+    // Must use a stable sort to preserve requested order of dimensions with duplicate strides
```

Comment:
std::sort or any ordered STL container doesn't work here? Or you need a custom sorting?

Reply:
During the sorting we need to permute the dimension order to match, so I think this simple sorting code on a small list should be fine.